module2/src/com/epam/decomposition/LogicBuilder.java

```diff
@@ -0,0 +1,93 @@
+package com.epam.decomposition;
+
+import com.epam.sorting.ArraySorter;
+
+public class LogicBuilder {
+
+    //task 1
+    public static int findLeastCommonMultiple(int firstNumber, int secondNumber) {
+        return ((firstNumber * secondNumber) / (SupportingLogicalBuilder.
+                findGreatestCommonDivisor(firstNumber, secondNumber)));
```

Comment:
You can move searching GCD from public method of other class to private method of current